### PR TITLE
DOC: Improve contibution guide

### PR DESCRIPTION
**Pull request by hoechenberger**
PR Description
--------------

Cleaned up parts of the contribution guide, rephrased a little, fixed a bit of formatting, and added instructions how to use non-global development installations of the `bids-validator` when running tests.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,45 +10,81 @@ it can be incorporated into the master branch.
 To help developing `mne-bids`, you will need a few adjustments to your
 installation as shown below.
 
-##### Running tests
+Running tests
+-------------
```
Comment (agramfort) [inline CONTRIBUTING.md:14]: this is a markdown file not an RST file. # should be used for titles

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -10,45 +10,81 @@ it can be incorporated into the master branch.
...
+If you want to run the tests with a development version of MNE-Python,
+you need to clone the `mne-python` repository and install it in "editable" mode
+locally.
 
+    $ pip uninstall mne  # First, uninstall PyPI-based mne-python
     $ git clone https://github.com/mne-tools/mne-python --depth 1
-    $ cd mne-python
-    $ pip uninstall mne  # uninstall pypi mne
-    $ pip install -e .  # use the cloned repo for a local install of mne
+    $ pip install -e ./mne-python  # install from the cloned repository
```
Comment (agramfort) [inline CONTRIBUTING.md:23]: if you don't have a proper clone I would suggest:

`pip install -U https://github.com/mne-tools/mne-python/archive/master.zip`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -10,45 +10,81 @@ it can be incorporated into the master branch.
...
+    $ pytest
+
+If you have installed the `bids-validator`
+on a per-user basis, set the environment variable `VALIDATOR_EXECUTABLE` to point to the path of the `bids-validator` before invoking `pytest`:
 
-##### Building the documentation
+    $ VALIDATOR_EXECUTABLE=../bids-validator/bids-validator/bin/bids-validator pytest
 
+Building the documentation
+--------------------------
```
Comment (agramfort) [inline CONTRIBUTING.md:87]: use # for a title

File: CONTRIBUTING.md (hunk 1 above)
Comment (hoechenberger) [reply CONTRIBUTING.md:14]: Actually both is valid MD: https://www.markdownguide.org/basic-syntax/#alternate-syntax
But I can switch to `#` for consistency

**Comment by agramfort** [comment]
please do :)

**Comment by hoechenberger** [comment]
@agramfort All comments addressed.

**Comment by agramfort** [comment]
thx @hoechenberger

**Pull request merged**